pyproject.toml

```diff
@@ -103,37 +87,60 @@ ignore = [
     # snake_case, such as a fixture that returns a mock of the FooBar class would
     # be named FooBar in CamelCase.
     "N",
-    # We are more lax about  comment formatting in the tests
-    "D",
-    
-    "PLR0913",
+    "PLR0913", # Too many arguments. Tests often have lots of arguments.
+    "PLR0917", # Too many positional arguments. Tests often have lots of arguments.
+    "PLR0904", # Too many public methods. Test classes often have lots of test methods.
+]
+"__init__.py" = [
+    "F401", # Ignore unused import errors on __init__ files to avoid having to add either a noqa stament or an __all__ declaration.
+]
 
-    # Lots of test methods don't use self, but we still want to group our tests
-    # into classes.
-    "PLR6301",
+[tool.coverage.run]
+branch = true
+parallel = true
+source = ["lms", "tests/unit"]
+omit = [
+    "*/lms/__main__.py",
+    "*/lms/scripts/init_db.py",
+    "lms/pshell.py",
+    "lms/migrations/*",
+    "lms/extensions/feature_flags/views/test.py",
+    "lms/views/feature_flags_test.py",
+]
+
+[tool.coverage.paths]
+source = ["src", ".tox/*tests/lib/python*/site-packages"]
 
-    "PLR0917", # too-many-arguments
-    "PLC2701", # private import
-    "PLR0904", # too-many-public-methods
-] 
-# Ignore unused import errors on __init__ files to avoid having to add either a noqa stament or an __all__ declaration.
-"__init__.py" = ["F401"]
+[tool.coverage.report]
+show_missing = true
+precision = 2
+fail_under = 100.00
+skip_covered = true
+exclude_also = [
+    # `if TYPE_CHECKING:` blocks are only executed while running mypy.
+    "if TYPE_CHECKING:",
+]
 
 
 [tool.mypy]
-python_version = 3.11
+allow_untyped_globals = true
+error_summary = false
+pretty = true
 warn_unused_configs = true
 warn_redundant_casts = true
 warn_unused_ignores = true
 check_untyped_defs = true
```

Comment:
This line is outside the template

I reckon we should do a `mypy/top_level/tail` entry point like we do for ruff here: https://github.com/hypothesis/cookiecutters/blob/main/_shared/project/pyproject.toml#L65C1-L68C1

@seanh Does that sound like the right approach?. 

It might be better to do the work to use `check_untyped_defs` everywhere but the entry point seems more practical right now.

Reply:
Sure, adding that include sounds like a good idea :+1: